src/nhc2-platform.ts

```diff
@@ -251,4 +283,29 @@ class NHC2Platform implements DynamicPlatformPlugin {
       });
     }
   }
+
+  private getSpeed(speed: FanSpeed) {
```

Comment:
I'd rename these methods to `fromFanSpeed` and `toFanSpeed` and move them to a static method class. You can add some basic testing as well for these methods. 

Testing is something we should take up as a whole for this plugin, should be easier since homebridge 1.2.0